### ISSUE TITLE
The Outlook calendar 'Calendar' previously configured for syncing is no longer available

### ISSUE THREAD

**Issue by gabrielmt-ven** [phw198/OutlookGoogleCalendarSync#1030]
> Checklist:-
> * [ ] [Upgrade](https://github.com/phw198/OutlookGoogleCalendarSync/wiki/Upgrading) to the latest release (alpha if available) ![latest release](https://img.shields.io/github/release/phw198/OutlookGoogleCalendarSync/all.svg)
> * [ ] Reproduce the issue
> * [ ] Complete the information below
> * [ ] Provide your `OGcalsync.log` file (see [how-to](https://github.com/phw198/OutlookGoogleCalendarSync/wiki/Reporting-problems#locating-your-ogcalsynclog-file))

**OGCS Version**: v2.8.4  
**Edition**: Portable  
**Sync Direction**:  O->G

**Bug Description**  
I got the following error after rebooting my WIN10 laptop

"The Outlook calendar 'Calendar' previously configured for syncing is no longer available.

'Calendar' calendar has been selected instead and any automated syncs have been temporarily disabled."

**Steps to Reproduce the Issue**  
1. Doble click on OutlookGoogleCalendarSync.exe
2. Application starts, shows the error, then crashes and got closed

![image](https://user-images.githubusercontent.com/63875745/85950293-4725a280-b95c-11ea-9b40-ea7edbc16ad0.png)

![image](https://user-images.githubusercontent.com/63875745/85950305-5278ce00-b95c-11ea-89ce-2eec0b5f6196.png)

**Any other information**  
Any additional information that may help troubleshoot the issue.  
:paperclip: Attach your `OGcalsync.log` file.

Last records from the log.  Exactly when application stared to crash
```
2020-06-28 16:15:29,383 INFO  1 OutlookGoogleCalendarSync.OutlookOgcs.OutlookNew [0] -  Disconnecting from Outlook application.
2020-06-28 16:15:31,867 DEBUG  15 OutlookGoogleCalendarSync.Forms.Splash [0] -  Disposed of splash screen.
2020-06-28 16:16:15,195 WARN  1 OutlookGoogleCalendarSync.Program [0] -  Tidying down any remaining Outlook references, as OGCS crashed out.
2020-06-28 16:16:15,199 WARN  1 OutlookGoogleCalendarSync.OutlookOgcs.Calendar [0] -  No Outlook process available to attach to.
2020-06-28 16:16:15,199 INFO  1 OutlookGoogleCalendarSync.OutlookOgcs.Calendar [0] -  Starting a new instance of Outlook.
2020-06-28 16:16:17,196 DEBUG  1 OutlookGoogleCalendarSync.OutlookOgcs.OutlookNew [0] -  Setting up Outlook connection.
2020-06-28 16:16:17,223 INFO  1 OutlookGoogleCalendarSync.OutlookOgcs.OutlookNew [0] -  Exchange server version: 15.20.3131.26
2020-06-28 16:16:17,223 INFO  1 OutlookGoogleCalendarSync.OutlookOgcs.OutlookNew [0] -  Exchange connection mode: olCachedDisconnected
2020-06-28 16:16:18,229 WARN  1 OutlookGoogleCalendarSync.OutlookOgcs.OutlookNew [0] -  >1s delay possibly due to Outlook security popup.
2020-06-28 16:16:18,229 DEBUG  1 OutlookGoogleCalendarSync.OutlookOgcs.OutlookNew [0] -  Getting active window inspector
2020-06-28 16:16:18,245 DEBUG  1 OutlookGoogleCalendarSync.OutlookOgcs.OutlookNew [0] -  Done.
2020-06-28 16:16:18,478 DEBUG  1 OutlookGoogleCalendarSync.OutlookOgcs.OutlookNew [0] -  Get the folders configured in Outlook
2020-06-28 16:16:18,480 DEBUG  1 OutlookGoogleCalendarSync.OutlookOgcs.OutlookNew [0] -  Finding default Mailbox calendar folders
2020-06-28 16:16:18,496 ERROR  1 OutlookGoogleCalendarSync.OGCSexception [0] -  System.ObjectDisposedException: Cannot access a disposed object.
Object name: 'TabPage'.
2020-06-28 16:16:18,496 ERROR  1 OutlookGoogleCalendarSync.OGCSexception [0] -  Code: 0x80131622;-2146232798
2020-06-28 16:16:18,496 ERROR  1 OutlookGoogleCalendarSync.OGCSexception [0] -     at System.Windows.Forms.Control.CreateHandle()
   at System.Windows.Forms.Control.get_Handle()
   at System.Windows.Forms.Control.CreateGraphicsInternal()
   at System.Windows.Forms.Control.CreateGraphics()
   at OutlookGoogleCalendarSync.OutlookOgcs.OutlookNew.findCalendars(Folders folders, Dictionary`2 calendarFolders, String excludeDeletedFolder, MAPIFolder defaultCalendar)
   at OutlookGoogleCalendarSync.OutlookOgcs.OutlookNew.getDefaultCalendar(NameSpace oNS, MAPIFolder& defaultCalendar)
2020-06-28 16:16:18,496 ERROR  1 OutlookGoogleCalendarSync.OGCSexception [0] -  System.ObjectDisposedException: Cannot access a disposed object.
Object name: 'TabPage'.
2020-06-28 16:16:18,496 ERROR  1 OutlookGoogleCalendarSync.OGCSexception [0] -  Code: 0x80131622;-2146232798
2020-06-28 16:16:18,496 INFO  1 OutlookGoogleCalendarSync.OutlookOgcs.Calendar [0] -  It appears Outlook has been restarted after OGCS was started. Reconnecting...
2020-06-28 16:16:18,496 INFO  1 OutlookGoogleCalendarSync.OutlookOgcs.Calendar [0] -  Attaching to the already running Outlook process.
2020-06-28 16:16:18,496 WARN  1 OutlookGoogleCalendarSync.OutlookOgcs.Calendar [0] -  Attachment failed.
2020-06-28 16:16:18,496 ERROR  1 OutlookGoogleCalendarSync.OGCSexception [0] -  System.ObjectDisposedException: Cannot access a disposed object.
Object name: 'TabPage'.
2020-06-28 16:16:18,496 ERROR  1 OutlookGoogleCalendarSync.OGCSexception [0] -  Code: 0x80131622;-2146232798
2020-06-28 16:16:18,496 INFO  1 OutlookGoogleCalendarSync.OutlookOgcs.Calendar [0] -  Starting a new instance of Outlook.
2020-06-28 16:16:18,962 DEBUG  1 OutlookGoogleCalendarSync.OutlookOgcs.OutlookNew [0] -  Setting up Outlook connection.
2020-06-28 16:16:18,962 INFO  1 OutlookGoogleCalendarSync.OutlookOgcs.OutlookNew [0] -  Exchange server version: 15.20.3131.26
2020-06-28 16:16:18,962 INFO  1 OutlookGoogleCalendarSync.OutlookOgcs.OutlookNew [0] -  Exchange connection mode: olCachedConnectedFull
2020-06-28 16:16:19,182 DEBUG  1 OutlookGoogleCalendarSync.OutlookOgcs.OutlookNew [0] -  Getting active window inspector
2020-06-28 16:16:19,184 DEBUG  1 OutlookGoogleCalendarSync.OutlookOgcs.OutlookNew [0] -  Done.
2020-06-28 16:16:19,480 DEBUG  1 OutlookGoogleCalendarSync.OutlookOgcs.OutlookNew [0] -  Get the folders configured in Outlook
2020-06-28 16:16:19,480 DEBUG  1 OutlookGoogleCalendarSync.OutlookOgcs.OutlookNew [0] -  Finding default Mailbox calendar folders
2020-06-28 16:16:19,497 ERROR  1 OutlookGoogleCalendarSync.OGCSexception [0] -  System.ObjectDisposedException: Cannot access a disposed object.
Object name: 'TabPage'.
2020-06-28 16:16:19,507 ERROR  1 OutlookGoogleCalendarSync.OGCSexception [0] -  Code: 0x80131622;-2146232798
2020-06-28 16:16:19,507 ERROR  1 OutlookGoogleCalendarSync.OGCSexception [0] -     at System.Windows.Forms.Control.CreateHandle()
   at System.Windows.Forms.Control.get_Handle()
   at System.Windows.Forms.Control.CreateGraphicsInternal()
   at System.Windows.Forms.Control.CreateGraphics()
   at OutlookGoogleCalendarSync.OutlookOgcs.OutlookNew.findCalendars(Folders folders, Dictionary`2 calendarFolders, String excludeDeletedFolder, MAPIFolder defaultCalendar)
   at OutlookGoogleCalendarSync.OutlookOgcs.OutlookNew.getDefaultCalendar(NameSpace oNS, MAPIFolder& defaultCalendar)
2020-06-28 16:16:19,507 WARN  1 OutlookGoogleCalendarSync.OGCSexception [0] -  Could not disconnect from Outlook.
2020-06-28 16:16:19,507 FAIL  1 OutlookGoogleCalendarSync.OGCSexception [0] -  System.ObjectDisposedException: Cannot access a disposed object.
Object name: 'TabPage'.
2020-06-28 16:16:19,507 FAIL  1 OutlookGoogleCalendarSync.OGCSexception [0] -  Code: 0x80131622;-2146232798
2020-06-28 16:16:19,524 INFO  1 OutlookGoogleCalendarSync.Program [0] -  Application closed.
```
Thanks in advance for your support

**Comment by phw198** [comment]
The first error related to this issue [occurred](https://github.com/phw198/OutlookGoogleCalendarSync/blob/dc15a668be518534f19b43bda65aacd22735a41a/src/OutlookGoogleCalendarSync/Forms/MainForm.cs#L213) before the log entries you have provided - please upload the full `OGcalsync.log` file one you have reproduced the issue. 

I note it looks like Outlook it disconnected? Does it make any difference if you ensure Outlook is connected first before starting OGCS?

**Comment by jay-gates** [comment]
FYI: This was happening to me for a couple of weeks, running portable 2.8.3. I tried starting OGCS before Outlook and also after Outlook was already fully running/connected, with the same result. I just downloaded portable v2.8.4, started it once with the original settings file (without issue, as expected), then replaced the settings file with a copy of the one from 2.8.3, and the problem did not reappear. I've attached a (slightly redacted) log from 2.8.3 showing the crash at the end.
[OGcalsync_redacted.log](https://github.com/phw198/OutlookGoogleCalendarSync/files/4961929/OGcalsync_redacted.log)

**Comment by gabrielmt-ven** [comment]
Hello I also upgraded to OGCS v2.8.4 and issue disappeared. I also copied the settings file from v2.8.3 into the folder from v.2.8.4, now I see all calendar entries in Google Calendar

Thanks for your support

Regards

**Comment by phw198** [comment]
@gabrielmt-ven Well that's annoying - you stated you were already on v2.8.4 in your OP and the didn't post your full logfile, so I couldn't tell that you weren't :roll_eyes: 

As always, upgrade to latest release before reporting issues...

**Comment by gabrielmt-ven** [comment]
Hello @phw198, sorry for that I started to have issues with v2.8.3 and then I started to look for information.  Initially upgrade didn't manage to solve the issue by downloading and installing version v2.8.4.  The reason was that I selected a different path to preserve the configuration file from v2.8.3 and I was using the shortcut pointing to all version.

The I saw the comment from @jay-gates  and reviewed again, so I decided to follow his workaround and it worked for me.

Sorry if it caused confusion.

Note: I didn't shared the full ,log file as it has personal data that I wasn't comfortable posting it in public forum.  Sorry for that, however I can share the part of the section were the error started to happen in a loop.  I hope it is helpful for you
```
2020-06-28 15:49:38,182 INFO  1 OutlookGoogleCalendarSync.Settings [0] -    Completed Syncs: 1279
2020-06-28 15:49:38,232 DEBUG  1 OutlookGoogleCalendarSync.Forms.Main [0] -  Create the timer for the auto synchronisation
2020-06-28 15:49:38,236 INFO  1 OutlookGoogleCalendarSync.Sync.SyncTimer [0] -  Schedule disabled.
2020-06-28 15:49:38,239 ERROR  1 OutlookGoogleCalendarSync.OGCSexception [0] -  System.InvalidOperationException: Window handle already exists.
2020-06-28 15:49:38,240 ERROR  1 OutlookGoogleCalendarSync.OGCSexception [0] -  Code: 0x80131509;-2146233079
2020-06-28 15:49:38,241 ERROR  1 OutlookGoogleCalendarSync.OGCSexception [0] -     at System.Windows.Forms.NativeWindow.CheckReleased()
   at System.Windows.Forms.NativeWindow.CreateHandle(CreateParams cp)
   at System.Windows.Forms.Control.CreateHandle()
   at System.Windows.Forms.Form.CreateHandle()
   at OutlookGoogleCalendarSync.Forms.Main..ctor(String startingTab)
   at OutlookGoogleCalendarSync.Program.Main(String[] args)
2020-06-28 15:49:38,241 FATAL  1 OutlookGoogleCalendarSync.Program [0] -  Application unexpectedly terminated!
2020-06-28 15:49:39,321 WARN  1 OutlookGoogleCalendarSync.Program [0] -  Tidying down any remaining Outlook references, as OGCS crashed out.
2020-06-28 15:49:39,321 DEBUG  1 OutlookGoogleCalendarSync.OutlookOgcs.OutlookNew [0] -  De-referencing all Outlook application objects.
2020-06-28 15:49:39,338 INFO  1 OutlookGoogleCalendarSync.OutlookOgcs.OutlookNew [0] -  Disconnecting from Outlook application.
2020-06-28 15:49:39,346 INFO  1 OutlookGoogleCalendarSync.Program [0] -  Application closed.
2020-06-28 15:50:11,021 INFO  1 OutlookGoogleCalendarSync.GoogleOgcs.ErrorReporting [0] -  Initialising error reporting to OFF
2020-06-28 15:50:11,030 INFO  1 OutlookGoogleCalendarSync.Program [0] -  Program started: v2.8.3.0
2020-06-28 15:50:11,030 INFO  1 OutlookGoogleCalendarSync.Program [0] -  Started interactively.
```

**Comment by phw198** [comment]
@gabrielmt-ven No worries - thanks for the clarification; can confirm this was a bug fixed in v2.8.4.
